public/app/features/alerting/unified/search/search.grammar

```diff
@@ -0,0 +1,40 @@
+@top AlertRuleSearch { expression+ }
+
+expression { (FilterExpression | FreeFormExpression) expression }
+
+FilterExpression {
+ DataSourceFilter |
+ NameSpaceFilter |
+ LabelFilter |
+ GroupFilter |
+ RuleFilter |
+ StateFilter |
+ TypeFilter
+}
+
+DataSourceFilter { filter<"ds" | "datasource"> }
```

Comment:
I don't think we should support these short-hand aliases and might add an additional maintenance burden. 

A few reasons I've mention in our review call;

* It assumes readers of the query know the keyword the short-hand refers to
* It adds additional code maintenance for them in the search code
* Adds additional complexity when we want to add additional features
	(ie. if we wanted to implement some sort of "type-ahead", we'd also have to make it work for these and the single character letters could add additional complexity to the code)